### PR TITLE
Improve logging around socket errors

### PR DESCRIPTION
**Pull request by puzpuzpuz**
#709 introduced duplicate logs in the case of socket errors, so this PR is aimed to get rid of duplicates and improve logging

### CODE REVIEW

File: src/network/ClientConnection.ts [hunk 1]
```diff
@@ -486,12 +486,19 @@ export class ClientConnection {
         let message = this.toString() + ' closed. Reason: ';
         if (this.closedReason != null) {
             message += this.closedReason;
-        } else if (this.closedCause != null) {
-            message += this.closedCause.name + '[' + this.closedCause.message + ']';
         } else {
             message += 'Socket explicitly closed';
         }
 
+        if (this.closedCause != null) {
```
Comment (mdumandag) [inline src/network/ClientConnection.ts:493]: Minor: If both `closedReason` and `closedCause` are not null, this is producing messages like `Reason: Failed to authenticate connection - Authentication failed. The configured cluster name..`. Maybe we could change those `connection.close` calls and make the produced log message better

**Review by mdumandag — APPROVED** [review]

File: src/network/ClientConnection.ts (hunk 1 above)
Comment (puzpuzpuz) [reply src/network/ClientConnection.ts:493]: Good suggestion. Tried to improve log messages in c8d38f8e73898482b02f080482939adb1e759255

Please take another look.

**Review by mdumandag — APPROVED** [review]

File: src/network/ClientConnection.ts (hunk 1 above)
Comment (mdumandag) [reply src/network/ClientConnection.ts:493]: Looks good, thanks

**Comment by puzpuzpuz** [comment]
@mdumandag thanks for the review!

**Pull request merged**